source/_posts/2017-11-07-which-hass-to-use.markdown

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: "Which installation should I do"
+description: "Should you use hass.io or VirtualEnv or hassbian?"
+date: 2017-11-07 10:00:00 +0000
+date_formatted: "November 07, 2017"
+author: Ryan McLean
+author_twitter: ryanm101
+comments: true
+categories: How-To
+---
+
+# Decisions, Decisions, Decisions...
+
+Ok you've made it this far you're interested in automating your home, but Home-assistant has you stuck there are 3 main ways to install it.. Which should you choose?
+
+*   Hass.io
+*   Python Virtual Environment
+*   Hassbian
+
+This brief article will hopefully give you a starting point to make your decision as all 3 are perfectly viable options. These are not prescriptive and hass.io for example can be installed on any linux server but if you are looking to do that, you don't need this article to tell you the options.
+
+<!--more-->
+
+## I want the quick answer with no Fluff.
+
+Simply choose where you fall into:
+
+*   I don't care I just want an autonomous home using my RPi / NUC -> Hass.io
+*   I don't want to fully dedicate my Pi and I want a simple install -> Hassbian
+*   I'm a developer -> Python Virtual Env
+
+## Considerations
+
+The primary consideration in my opinion is HOW do you want to use Home-assistant?
+
+In my case I want a simple stable install for my "production" instance and I have a few spare Raspberry Pis, So I use Hass.io for that "production" instance.
+
+On my Linux Laptop I use the Python Virtual Environment install so I can test different configs and I also have multiple Virtual Environments on the laptop allowing me to develop components using different versions of the libraries.
+
+## Install Options
+### Option 1: [Hass.io](https://home-assistant.io/hassio/)
+
+This is the simplest option, it hides all that nasty code from you so can only edit the config files and all you have to do is click update in the UI and hey presto you have an updated install.
+
+This is the best option for beginners or people that want that appliance "just works feel".
+
+For those that like to get into the nitty gritty and tinker, it's the worst option.
+
+### Option 2: [Hassbian](https://home-assistant.io/docs/hassbian/installation/)
+
+You have a raspberry Pi, you want to be able to see and edit the internals but you want a simple install and you want to use your RPi for multiple things.
+
+Simply download the image and install on the SD card, you can now edit the configs and also tweak code etc as needed.
+
+This option is basically Option3 for lazy people :)
```

Comment:
I'd mention that hass.io has certain limitations and that you could likely be prevented from doing something you'd normally be able to do in Linux